src/thor/alternates.cc

```diff
@@ -175,20 +169,19 @@ bool validate_alternate_by_sharing(std::vector<std::unordered_set<GraphId>>& sha
 
     // if an edge on the candidate_path is encountered that is also on one of the existing paths,
     // we count it as a "shared" edge
-    float shared_length = 0.f, total_length = 0.f;
+    float shared_length = 0.f;
     for (const auto& cpi : candidate_path) {
       const auto length = &cpi == &candidate_path.front()
                               ? cpi.path_distance
                               : cpi.path_distance - (&cpi - 1)->path_distance;
-      total_length += length;
       if (shared.find(cpi.edgeid) != shared.end()) {
         shared_length += length;
       }
     }
 
-    // throw this alternate away if it shares more than at_most_shared with any of the chosen paths
-    assert(total_length > 0);
-    if ((shared_length / total_length) > at_most_shared) {
+    // throw this alternate away if any of the chosen paths shares more than at_most_shared with it
+    assert(paths[i].back().path_distance > 0);
+    if ((shared_length / paths[i].back().path_distance) > at_most_shared) {
```

Comment:
this is  important change. What was before:
"sharing value is a part of shared edges of an alternative candidate with some already chosen route"
now is
"sharing value is a part of shared edges of some already chosen route with an alternative candidate".

Reply:
Let's consider a particular example:
```
A------------------------------------B---------C----------------------------------------D
                                     |         |
                                     |         |
                                     |         |
                                     |         |
                                     E---------F
```
here `ABCD` is an optimal route and `ABEFCD` is an alternative candidate.
Let's assume that `len(AB + CD) / len(ABCD) = 0.85` and `len(AB + CD) / len(ABEFCD) = 0.75`. Current **master** will allow `ABEFCD` to be an alternative despite the fact that `BEFC` detour is obviously unreasonable. This **branch** will reject this alternative due to the fact that `sharing = 0.85 > 0.75 = at_most_shared` value.

Reply:
this was what was originally intended actually... good catch! i should also note though that in another pr we are looking into problems with teh path_distance not being set on the last edge label, which is where the pathinfo gets it. it is my h unch that this is in the case that we find a path using only the forward search tree but i havent proven it yet.

Reply:
Hm, thanks for the comment,  I will check it

Reply:
@kevinkreiser replaced division with multiplication to avoid possible problems